kratos/utilities/force_and_torque_utils.cpp

```diff
@@ -0,0 +1,75 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+
+/* System includes */
+
+/* External includes */
+
+/* Project includes */
+#include "utilities/force_and_torque_utils.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+#include "utilities/math_utils.h"
+
+namespace Kratos
+{
+
+
+std::array<array_1d<double,3>,2> ForceAndTorqueUtils::SumForceAndTorque(const ModelPart& rModelPart,
+                                                                        const array_1d<double,3>& rReferencePoint)
+{
+    KRATOS_TRY
+
+    using CompoundArray = array_1d<double,6>;
+
+    CompoundArray force_and_moment = block_for_each<SumReduction<CompoundArray>>(
+        rModelPart.GetCommunicator().LocalMesh().Nodes(),
+        [&rReferencePoint](const Node<3>& rNode) -> CompoundArray
+        {
+            array_1d<double,3> force, moment;
```

Comment:
These should be allocated outside and passed inside the `block_for_each` using the Thread Local Storage (TLS).

Reply:
Well, as far as I know, ```array_1d``` is stack-allocated and similar to ```std::array```, so their allocation should just be a stack pointer increment. On the other hand, objects in TLS have to be dereferenced, which isn't completely free either.

It's not clear to me which one's cheaper, and it looks like this varies across compilers as well:
https://godbolt.org/z/MsWx7P5E5

(of course I'm assuming that ```array_1d``` is more or less identical to ```std::array```)

Reply:
(I think your assumption is fair enough)

Interesting. After having played around with your test (thanks for setting it up), I agree that allocating stack arrays inside the lambda bodies is not that bad, specially when complex TLS containers need to be set.

Thanks!